velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp

```diff
@@ -918,3 +918,42 @@ TEST_F(ParquetReaderTest, testEmptyRowGroups) {
 
   assertReadWithReaderAndExpected(fileSchema, *rowReader, expected, *leafPool_);
 }
+
+TEST_F(ParquetReaderTest, structSkipRowGroup) {
+  // single_row_struct.parquet has 2 columns (s.a: INT64 and s.b: INT64).
+  // It has 1 row.
+  // It has one row_group.
+  // Following is the data in the file
+  // +------------------+
+  // | s                |
+  // |------------------|
+  // | {'a': 0, 'b': 1} |
+  // +------------------+
+
+  RuntimeStatistics runtimeStats_ = dwio::common::RuntimeStatistics();
+  const std::string upper(getExampleFilePath("single_row_struct.parquet"));
+
+  facebook::velox::dwio::common::ReaderOptions readerOptions{leafPool_.get()};
+  readerOptions.setFileColumnNamesReadAsLowerCase(true);
+  RowReaderOptions rowReaderOpts;
+  auto rowType = ROW({"s.a"}, {BIGINT()});
+
+  // Reader without filter
+  auto reader1 = createReader(upper, readerOptions);
```

Comment:
Let's use names with meanings instead of 1 or 2.  E.g. reader1 -> readerWithoutFilter

Reply:
Done